### PR TITLE
Husky to check for prettier and eslint

### PR DESCRIPTION
**Pull request by iShelar**
Issue: #2 

Changes: Lint & Prettier with pre commit configuration.

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import { Footer } from "../components";
+import React from "react";
```
Comment (supminn) [inline pages/_app.js]: Why has `React` been imported unnecessarily?

File: pages/_document.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { Html, Head, Main, NextScript } from "next/document";
+import React from "react";
```
Comment (supminn) [inline pages/_document.js]: Why has `React` been imported unnecessarily?

File: package.json [hunk 3]
```diff
@@ -17,10 +22,23 @@
   },
   "devDependencies": {
     "autoprefixer": "^10.4.12",
-    "eslint": "8.24.0",
+    "eslint": "^8.0.1",
```
Comment (supminn) [inline package.json]: Is there a reason for using a lower version of `eslint`?

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  env: {
+    browser: true,
+    es2021: true,
+  },
+  extends: ["plugin:react/recommended", "standard", "prettier"],
```
Comment (supminn) [inline .eslintrc.js]: Could you check if there is a plugin for `nextjs`. Check [this](https://nextjs.org/docs/basic-features/eslint) out.
I believe we should be using `eslint-plugin-next`

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  overrides: [],
+  parserOptions: {
+    ecmaVersion: "latest",
+    sourceType: "module",
+  },
+  plugins: ["react"],
```
Comment (supminn) [inline .eslintrc.js]: We need to replace it with `next`

File: package.json [hunk 6]
```diff
@@ -17,10 +22,23 @@
...
     "eslint-config-next": "12.3.1",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-config-standard": "^17.0.0",
+    "eslint-plugin-import": "^2.25.2",
+    "eslint-plugin-n": "^15.0.0",
+    "eslint-plugin-promise": "^6.0.0",
+    "eslint-plugin-react": "^7.31.8",
```
Comment (supminn) [inline package.json]: Could you check if there is a plugin for `nextjs`. Check [this](https://nextjs.org/docs/basic-features/eslint) out.
I believe we should be using `eslint-plugin-next`

File: .eslintrc.js [hunk 7]
```diff
@@ -10,5 +10,7 @@ module.exports = {
     sourceType: "module",
   },
   plugins: ["react"],
-  rules: {},
+  rules: {
+    "react/prop-types": "off",
```
Comment (supminn) [inline .eslintrc.js]: Why have we set this property to `off`?

**Review by supminn — CHANGES_REQUESTED** [review]
Hi @iShelar
Thank you for your contribution. I have added a few comments for clarification. Could you get them updated?

File: .eslintrc.js (hunk 5 above)
Comment (iShelar) [reply .eslintrc.js]: Oops. Sorry bad mistake. Fixing it.

File: .eslintrc.js (hunk 7 above)
Comment (iShelar) [reply .eslintrc.js]: Actually I setup eslint using "react" plugin. So it was asking me to define prop-types. Now next plugin is added. It is working fine. Thanks.

File: package.json (hunk 3 above)
Comment (iShelar) [reply package.json]: No. I have updated the version. Thanks

File: package.json (hunk 6 above)
Comment (iShelar) [reply package.json]: Thanks. It's working fine now.

**Comment by iShelar** [comment]
Hi @supminn, Thanks for review. The comments are addressed.

File: .eslintrc.js (hunk 4 above)
Comment (iShelar) [reply .eslintrc.js]: Thanks Supriya. I have updated it to next.

File: pages/_document.js (hunk 2 above)
Comment (iShelar) [reply pages/_document.js]: May be because I was using react plugin. I have updated the plugin & remove this extra imports. Thanks

**Review by supminn — APPROVED** [review]
LTGM.
Changes approved.

**Pull request merged**